features/nanostack/mbed-mesh-api/mbed-mesh-api/WisunInterface.h

```diff
@@ -69,6 +91,253 @@ class WisunInterface : public MeshInterfaceNanostack {
      * */
     mesh_error_t set_network_regulatory_domain(uint8_t regulatory_domain = 0xff, uint8_t operating_class = 0xff, uint8_t operating_mode = 0xff);
 
+    /**
+     * \brief Get Wi-SUN network regulatory domain, operating class and operating mode.
+     *
+     * Function reads regulatory_domain, operating_class and operating_mode from mbed-mesh-api.
+     *
+     * \param regulatory_domain Values defined in Wi-SUN PHY-specification.
+     * \param operating_class Values defined in Wi-SUN PHY-specification.
+     * \param operating_mode Values defined in Wi-SUN PHY-specification.
+     * \return MESH_ERROR_NONE on success.
+     * \return MESH_ERROR_UNKNOWN in case of failure.
+     * */
+    mesh_error_t get_network_regulatory_domain(uint8_t *regulatory_domain, uint8_t *operating_class, uint8_t *operating_mode);
+
+    /**
+     * \brief Validate Wi-SUN network regulatory domain, operating class and operating mode.
+     *
+     * Function validates regulatory_domain, operating_class and operating_mode. Function can be used to test that values that will
+     * be used on set function are valid.
+     *
+     * \param regulatory_domain Values defined in Wi-SUN PHY-specification.
+     * \param operating_class Values defined in Wi-SUN PHY-specification.
+     * \param operating_mode Values defined in Wi-SUN PHY-specification.
+     * \return MESH_ERROR_NONE on success.
+     * \return MESH_ERROR_UNKNOWN in case of failure.
+     * */
+    mesh_error_t validate_network_regulatory_domain(uint8_t regulatory_domain, uint8_t operating_class, uint8_t operating_mode);
+
+    /**
+     * \brief Set Wi-SUN network size.
+     *
+     * Function stores new parameters to mbed-mesh-api and uses them when connect() is called next time.
+     * If device is already connected to the Wi-SUN network then device will restart network discovery after
+     * changing the network size.
+     *
+     * Default value: medium
+     * Small network size: less than hundred devices
+     * Medium network size: hundreds of devices
+     * Large network size: thousands of devices
+     * Automatic: when discovering the network, network size is learned from advertisements and timings adjusted accordingly
+     * Certificate: used on testing
```

Comment:
I don't think we should limit the network size in this interface by enum it could be from 0 to 20 or something
Then just assign values for those points that we have inside stack
Otherwise changing this later on becomes really hard

Also Automatic is not valid option and should not be here
0 == 25 less
1 == 100
2== 200
10 == 1000
20 == 10000

Reply:
Nanostack interface has these fixed values (and only those are supported):
#define NETWORK_SIZE_CERTIFICATE    0x00
#define NETWORK_SIZE_SMALL          0x01
#define NETWORK_SIZE_MEDIUM         0x08
#define NETWORK_SIZE_LARGE          0x10
#define NETWORK_SIZE_AUTOMATIC      0xFF
so will I add a conversion to mesh api so that 0 --> 6 means small, 7 - 12 medium, 13 to 18 large etc...

Reply:
Changed network size to hundreds of devices, removed defines.

Reply:
I don't like that automatic. It was causing more problems at nano stack side.